From: Fab Stz <fabstz-it@yahoo.fr>
Date: Sat, 15 Mar 2025 09:03:11 +0100
Subject: Set wx-config version

Origin: Debian
Forwarded: https://freefilesync.org/forum/viewtopic.php?t=9685
Last-Update: 2022-08-12

Useful when there are multiple versions of wx-config installed on the system
---
 FreeFileSync/Source/Makefile              | 5 +++--
 FreeFileSync/Source/RealTimeSync/Makefile | 5 +++--
 2 files changed, 6 insertions(+), 4 deletions(-)

diff --git a/FreeFileSync/Source/Makefile b/FreeFileSync/Source/Makefile
index 3a180ce..fc2576b 100644
--- a/FreeFileSync/Source/Makefile
+++ b/FreeFileSync/Source/Makefile
@@ -2,12 +2,13 @@ CXX ?= g++
 exeName = FreeFileSync_$(shell arch)
 
 PKG_CONFIG=pkgconf
+WX_CONFIG=wx-config --version=3.2
 
 CXXFLAGS += -std=c++23 -pipe -DWXINTL_NO_GETTEXT_MACRO -I../.. -I../../zenXml -include "zen/i18n.h" -include "zen/warn_static.h" \
            -Wall -Wfatal-errors -Wmissing-include-dirs -Wswitch-enum -Wcast-align -Wnon-virtual-dtor -Wno-unused-function -Wshadow -Wno-maybe-uninitialized \
-           -O3 -DNDEBUG `wx-config --cxxflags --debug=no` -pthread
+           -O3 -DNDEBUG `$(WX_CONFIG) --cxxflags --debug=no` -pthread
 
-LDFLAGS += -s `wx-config --libs std, aui, richtext --debug=no` -pthread
+LDFLAGS += -s `$(WX_CONFIG) --libs std, aui, richtext --debug=no` -pthread
 
 
 CXXFLAGS  += `$(PKG_CONFIG) --cflags openssl`
diff --git a/FreeFileSync/Source/RealTimeSync/Makefile b/FreeFileSync/Source/RealTimeSync/Makefile
index c2ac495..69b8756 100644
--- a/FreeFileSync/Source/RealTimeSync/Makefile
+++ b/FreeFileSync/Source/RealTimeSync/Makefile
@@ -2,12 +2,13 @@ CXX ?= g++
 exeName = RealTimeSync_$(shell arch)
 
 PKG_CONFIG=pkgconf
+WX_CONFIG=wx-config --version=3.2
 
 CXXFLAGS += -std=c++23 -pipe -DWXINTL_NO_GETTEXT_MACRO -I../../.. -I../../../zenXml -include "zen/i18n.h" -include "zen/warn_static.h" \
            -Wall -Wfatal-errors -Wmissing-include-dirs -Wswitch-enum -Wcast-align -Wnon-virtual-dtor -Wno-unused-function -Wshadow -Wno-maybe-uninitialized \
-           -O3 -DNDEBUG `wx-config --cxxflags --debug=no` -pthread
+           -O3 -DNDEBUG `$(WX_CONFIG) --cxxflags --debug=no` -pthread
 
-LDFLAGS += -s `wx-config --libs std, aui, richtext --debug=no` -pthread
+LDFLAGS += -s `$(WX_CONFIG) --libs std, aui, richtext --debug=no` -pthread
 
 
 CXXFLAGS += `$(PKG_CONFIG) --cflags gtk+-3.0`
